### ISSUE TITLE
Add support to compaction property to enable/disable background compaction

### ISSUE THREAD

**Issue by raphaelsc** [scylladb/scylla#1488]
Details of the property:

name: enabled   default value: true Set to enable background compaction.

**Comment by tzach** [comment]
@raphaelsc how and why should the user choose to enable/disable?
Its look like something Scylla should take care for

**Comment by raphaelsc** [comment]
@tzach: In scylla, there's currently no easy way of disabling regular compaction. User could set compaction strategy of column family to NullCompactionStrategy. There's a nodetool command that allows to enable or disable it, but it's currently not supported by us. In my opinion, user shouldn't disable it or read performance will suffer, but I suppose user may need to do it for analyzing impact of compaction on performance or something.

**Comment by tzach** [comment]
Thanks @raphaelsc make sense
Let's wait for user requests before moving forward

**Comment by jeremysears** [comment]
As described in #1808, for debugging purposes, it would be nice to have an equivalent for `nodetool disableautocompaction` and `nodetool enableautocompaction`.

**Comment by slivne** [comment]
@avikivity if I am not mistaken you had reservation around allowing a user to control this - is that still true, or can we move forward and add support for this ?

**Comment by avikivity** [comment]
It's easy to kill a cluster or kill read performance by disabling compaction.

It's okay for debugging purposes or when the user knows what they are doing.

**Comment by glommer** [comment]
To be clear, I am fine if we don't support *this* specific interface.
Turns out we have a way to disable compactions on a global level, by setting the strategy to NullCompactionStrategy.

This specific interface is maybe more convenient because it allows you to come back to your strategy more easily.

What is *really* crucial in my view is #440. The reason for that is that if we are disabling compaction because things are going south, chances are we won't be able to make schema changes.

(I ended up bumping priority of both "disable compaction" issues - this one is my bad).
I will decrease this one and I will leave up to you whether we should enable this particular interface or not

**Comment by slivne** [comment]
@bhalevy compaction related - can be placed in the background (or for someone that wants to learn compaction)

**Comment by sitano** [comment]
This patch introduces an API point /column_family/autocompaction/{name} that supports GET and POST methods to read and control column family minor compaction.

It would be nice to receive some feedback before going further with it.

I am concerned about whether the GET `/column_family/autocompaction/{name}` must depend on the schema null compaction strategy. In my current implementation, it depends. Because AFAIU semantically null CS literally means - do not compact, and thus we can count that fact as a result. 

Another issue with this handle is that as far as the user can change CS without changing the schema, setting CS must reset auto compaction flag to default. Maybe I am missing something here.

**Comment by bhalevy** [comment]
> This patch introduces an API point `/column_family/autocompaction/{name}` that supports GET and POST methods to read and control column family minor compaction.
> 
> It would be nice to receive some feedback before going further with it.
> 
> I am concerned about whether the GET `/column_family/autocompaction/{name}` must depend on the schema null compaction strategy. In my current implementation, it depends. Because AFAIU semantically null CS literally means - do not compact, and thus we can count that fact as a result.
> 
> Another issue with this handle is that as far as the user can change CS without changing the schema, setting CS must reset auto compaction flag to default. Maybe I am missing something here.
> 
> p.s. I did not use original Glauber implementation which changes CF in parallel.

ping @raphaelsc

**Comment by raphaelsc** [comment]
@bhalevy that suggested approach is outdated, we moved towards a different solution. please check out https://github.com/scylladb/scylla/pull/5877/
